packages/common/test/i18n/format_date_spec.ts

```diff
@@ -378,5 +378,11 @@ describe('Format date', () => {
          expect(formatDate('2010-01-02', 'YYYY', 'en')).toEqual('2009');
          expect(formatDate('2010-01-04', 'YYYY', 'en')).toEqual('2010');
        });
+
+    // https://github.com/angular/angular/issues/40377
+    it('should format date with year between 0 and 99 correctly', () => {
+      expect(formatDate('0099-01-11T12:00:00.014-0500', defaultFormat, ɵDEFAULT_LOCALE_ID))
```

Comment:
Does this test actually fail without the fix?
https://stackblitz.com/edit/angular-ivy-qhzmxz?devtoolsheight=33&file=src/app/app.component.ts

Reply:
I think you actually don't want to pass in an ISO formatted date, since that will actually get parsed correctly!!!  Instead this should be:

```suggestion
      expect(formatDate('0099-01-11', defaultFormat, ɵDEFAULT_LOCALE_ID))
```

See https://stackblitz.com/edit/angular-ivy-mxrnbk

Reply:
Actually it fails when you use `"YYYY"` as format, I will change `defaultFormat` to `"YYYY"`

Reply:
https://stackblitz.com/edit/angular-ivy-lbeum3?devtoolsheight=33&file=src/app/app.component.ts